File: cs/labs/cpp/hello_world_test.gpt.cc
// cs/labs/cpp/hello_world_test.gpt.cc
// cs/labs/cpp/hello_world_test.cc
#include <cassert>

#include "hello.h"

int main() {
  auto msg = SayHello("World");
  assert(msg == "Hello World!");
  return 0;
}

File: cs/labs/csharp/hello_world_test.gpt.cs
using System;
using Labs.CSharp;

namespace Labs.CSharp;

public class HelloWorldTest
{
    public static int Main()
    {
        var result = Greeter.SayHello("World");
        var expected = "Hello World!";
        
        if (result != expected)
        {
            Console.Error.WriteLine($"Expected: '{expected}', Got: '{result}'");
            return 1;
        }
        
        Console.WriteLine("Test passed!");
        return 0;
    }
}

